### PR TITLE
Clarify error message if duotone color values is incorrect

### LINKED ISSUE #50200: error warning it site editor with duotones
### Description

I do have custom duotones in theme.json, they work fine

this is the definition btw

```
            "duotone": [
                {
                    "name": "DT Powerbox 200 to 300",
                    "slug": "dt-powerbox-200-300",
                    "colors": [
                        "var(--wp--preset--color--powerbox-200)",
                        "var(--wp--preset--color--powerbox-300)"
                    ]
                },
                {
                    "name": "DT Powerbox 700 to 800",
                    "slug": "dt-powerbox-700-800",
                    "colors": [
                        "var(--wp--preset--color--powerbox-700)",
                        "var(--wp--preset--color--powerbox-800)"
                    ]
                }
			],
```
			
			
But when i do use gutenberg plugin and having debug=true theese errors popup when I open Site Editor:

<b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>344</b><br />
 <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>345</b><br />
  | <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>346</b><br />
  | <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>347</b><br />

### Step-by-step reproduction instructions

debug = true
set up duotones in theme.json
enable gutenberg plugin
go to site editor

### Screenshots, screen recording, code snippet

_No response_

### Environment info

latest versions of guenberg and wordpress

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes: #50200
Similar to #50714

## What?
This PR fixes a PHP warning error that appears when the duotone color is not HEX or RGB in theme.json and changes the text to be more specific.

## Why?
In theme.json, `settings.color.duotone[].colors` values support only strings representing HEX or RGB. If any other value is present, it cannot be parsed into the RGBA value and will refer to a non-existent index in the array, causing a PHP Warning error.

The fact that only RGB and HEX are allowed is also [defined](https://github.com/WordPress/gutenberg/blob/493d95afd41e37ba8385b6420d8023b11c413dd8/schemas/json/theme.json#L170) as `description` in the theme.json schema.

description

However, many developers may not be aware of this because (in my VSCode) I have to mouse over the value to read that description. I thought it would make sense to let developers know what values are incorrect by means of more explicit error messages.

## How?
As in #50714, the _doing_it_wrong function indicates color strings that could not be parsed. Also, the invalid value is not used as a value for the duotone filter.

## Testing Instructions

Activate the Empty Theme theme and update theme.json as follows:

```
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"settings": {
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		},
		"color": {
			"duotone": [
				{
					"name": "Black To White",
					"slug": "black-to-white",
					"colors": [
						"var(--wp--preset--color--black)",
						"#fff"
					]
				},
				{
					"name": "Pale Pink to Vivid Red",
					"slug": "pale-pink-to-vivid-red",
					"colors": [
						"#f78da7",
						"var(--wp--preset--color--vivid-red)"
					]
				},
				{
					"name": "Incorrect Values",
					"slug": "incorrect",
					"colors": [
						"incorrect-value-1",
						"incorrect-value-2"
					]
				},
				{
					"name": "Pale Cyan Blue to Vivid Purple",
					"slug": "pale-cyan-blue-to-vivid-purple",
					"colors": [
						"#8ed1fc",
						"#9b51e0"
					]
				}
			]
		}
	}
}
```

Access the editor and verify that the error message changes.

### Before

```
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 533
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 534
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 535
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 536
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 533
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 534
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 535
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 536
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 533
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 534
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 535
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 536
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 533
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 534
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 535
PHP Warning:  Trying to access array offset on value of type null in /var/www/html/wp-content/plugins/gutenberg/lib/class-wp-duotone-gutenberg.php on line 536


```

### After

```
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "var(--wp--preset--color--black)" is not a hex or rgb string in theme.json settings. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "var(--wp--preset--color--vivid-red)" is not a hex or rgb string in theme.json settings. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "incorrect-value-1" is not a hex or rgb string in theme.json settings. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "incorrect-value-2" is not a hex or rgb string in theme.json settings. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
```

### CODE REVIEW

File: lib/class-wp-duotone-gutenberg.php [hunk 1]
```diff
@@ -530,10 +530,19 @@ private static function get_filter_svg( $filter_id, $colors ) {
...
 			$color = self::colord_parse( $color_str );
 
-			$duotone_values['r'][] = $color['r'] / 255;
-			$duotone_values['g'][] = $color['g'] / 255;
-			$duotone_values['b'][] = $color['b'] / 255;
-			$duotone_values['a'][] = $color['a'];
+			if ( null === $color ) {
+				$error_message = sprintf(
+					/* translators: %s: duotone colors */
+					__( '"%s" is not a hex or rgb string in theme.json settings.', 'gutenberg' ),
```
Comment (glendaviesnz) [inline lib/class-wp-duotone-gutenberg.php]: Tested well for me. I wonder if for even more clarity we should say `'"%s" in theme.json settings.color.duotone is not a hex or rgb string.'?

Comment (t-hamano) [reply lib/class-wp-duotone-gutenberg.php]: Thanks for the review! Updated with dae5201. Here are the new error messages:

```
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "var(--wp--preset--color--black)" in theme.json settings.color.duotone is not a hex or rgb string. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "var(--wp--preset--color--vivid-red)" in theme.json settings.color.duotone is not a hex or rgb string. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "incorrect-value-1" in theme.json settings.color.duotone is not a hex or rgb string. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896
PHP Notice:  Function WP_Duotone_Gutenberg::get_filter_svg was called <strong>incorrectly</strong>. "incorrect-value-2" in theme.json settings.color.duotone is not a hex or rgb string. Please see <a href="https://wordpress.org/documentation/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 6.3.0.) in /var/www/html/wp-includes/functions.php on line 5896

```

**Review by glendaviesnz — APPROVED** [review]
LGTM

**Comment by t-hamano** [comment]
I believe this PR needs to be backported to WordPress core. ~~I will add a comment to #51077 when the ticket and PR are ready in WordPress core.~~ I learned that numerous changes were made to the class-wp-duotone-gutenberg.php file. Therefore, I only tied the PR to the target file in #51077.

**Comment by ramonjd** [comment]
> I learned that numerous changes were made to the class-wp-duotone-gutenberg.php file. Therefore, I only tied the PR to the target file in https://github.com/WordPress/gutenberg/issues/51077.

Thanks @t-hamano !

**Pull request merged**